custom_registration/models.py

```diff
@@ -61,6 +65,24 @@ def check_user_token(user, token):
 def confirm_user_email(user):
     user.profile.email_confirmed_at = timezone.now()
     user.profile.save()
+    generate_team_invitation_by_email_domain(user)
+
+
+def generate_team_invitation_by_email_domain(user):
+    domain = re.search("@([\w.]+)$", user.email).groups()[0]
```

Comment:
This regular expression won't work with all valid email addresses. Ex. email@t-online.de

You should use the following:
```python 
domain = re.search(
    r'((?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)+)(?:[A-Z0-9-]{2,63}(?<!-))\Z',
    user.email,
    re.IGNORECASE
).group()
```
I borrowed the RE from [django's email validator](https://github.com/django/django/blob/master/django/core/validators.py#L171)